### PR TITLE
Automated Data Logging

### PR DESCRIPTION
**Pull request by AroshHeenkenda**
## Description
These changes allow for automated data logging into an SQLite database each time a true `v3/start` message is received and subsequent conversion into an excel file when a false message is received. The excel files are stored in the `dashboard/server/data` folder to be uploaded onto Google Drive.

Updates have also been made to the relevant `README` and `env` files.

## Screenshots
_Screenshot 1_
image

_Screenshot 2_
image

_Screenshot 3_
image

_Screenshot 4_
image

_Screenshot 5_
image

_Screenshot 6_
image

## Steps to Test

1. In a terminal navigate to the correct directory to `data-acquisition-system/DAS/das`.
2. Set up your `.env` file accordingly.
    - `MQTT_LOG_FILE` should be your local path that leads to the `data-acquisition-system/DAS/das/sqlite/` folder
    - `EXCEL_LOG_FILE` should be your local path that leads to the `dashboard/server/data/` folder.
3. Spawn a poetry shell using `poetry env use 3.7`.
    - **Note if this doesn't work use `poetry env use PATH_TO_PY3.7`.**
4. Open the poetry shell and run `poetry install`.
5. Open a split terminal navigating to `data-acquisition-system/DAS/das`, and open a poetry shell.
6. In the first terminal run `python log_2_excel.py`, you may use any command line arguments but take note of the `host`.
7. In the second terminal run `python V3_fake_module.py`.
   - You should notice that a true start message is received and an SQLite database should appear in the `sqlite` folder, as per _Screenshots 1 & 2_.
8. Stop the V3_fake_module by running `CTRL + C` in that terminal.
   - A false start message should be received, and an excel file should be saved in the correct folder, as per _Screenshots 3 & 4_.
   - Check that the data has been correctly logged into the excel file by opening it
9. To test that we can only start if the logger hasn't started and stop when the logger has started follow the below:
   - In a terminal run `python log_2_excel.py`.
   - Send a false start message, `mosquitto_pub -t "v3/start" -m '{"start": false}'`, and check the terminal shows something similar to _Screenshot 5_.
   - Send a true start message, `mosquitto_pub -t "v3/start" -m '{"start": true}'`, and then send it again, check the terminal shows something similar to _Screenshot 6_.

For extra testing, try sending messages using different brokers such as the one on the primary display. Testing can also be done by switching to this branch on the primary display and running the code on there to check that logs are saved correctly.

### CODE REVIEW

File: DAS/das/log_2_excel.py [hunk 1]
```diff
@@ -2,159 +2,400 @@
...
+        self.logging = False
+        self.MQTT_LOG_FILE = db_file 
+        self.EXCEL_LOG_FILE = xl_file + fname
+        self.time = "" 
+
+        self.recorder = None
+
+
+    def on_connect(self, client, userdata, flags, rc):
+        """Callback for when cleint receives a CONNNACK response."""
```
Comment (jmark-9) [inline DAS/das/log_2_excel.py]: Tested on my computer and worked. Just this typo otherwise looks good

**Comment by jmark-9** [comment]
Make sure to pull from the master branch to get the latest changes to the data acquisition system in this branch (i.e. the crash alert code). Also noticed that the SQLite database files appear in the `das` folder rather than the `sqlite` folder so that could be something to look into.

**Comment by jmark-9** [comment]
Maybe if you could more clearly specify the path to the folders that the excel and sql files are meant to go in so that it's easier to set up the .env file. Perhaps in the README or in this pr's description.

**Comment by AroshHeenkenda** [comment]
> Maybe if you could more clearly specify the path to the folders that the excel and sql files are meant to go in so that it's easier to set up the .env file. Perhaps in the README or in this pr's description.

Hopefully the README and PR have clearer setup instructions.

**Comment by jmark-9** [comment]
> > Maybe if you could more clearly specify the path to the folders that the excel and sql files are meant to go in so that it's easier to set up the .env file. Perhaps in the README or in this pr's description.
> 
> Hopefully the README and PR have clearer setup instructions.

Looks good to me now 👍

**Review by jmark-9 — APPROVED** [review]

**Comment by KIR007-glitch** [comment]
Looks good to me everything is working

**Pull request merged**